src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -0,0 +1,177 @@
+    Ref,
+    SegmentGenerator,
+    Anything,
+    Indent,
+    Dedent,
+    Nothing,
+    OptionallyBracketed,
+    StringLexer,
+    RegexLexer,
+    CodeSegment,
+    CommentSegment,
+    WhitespaceSegment,
+    NewlineSegment,
+    StringParser,
+    NamedParser,
+    RegexParser,
+    Conditional,
+)
+
+from sqlfluff.core.dialects.base import Dialect
+from sqlfluff.core.dialects.common import AliasInfo
+from sqlfluff.core.parser.segments.base import BracketedSegment
+
+from sqlfluff.dialects.ansi_keywords import (
+    ansi_reserved_keywords,
+    ansi_unreserved_keywords,
+)
+from sqlfluff.dialects.tsql_keywords import (
+    # BARE_FUNCTIONS,
+    RESERVED_KEYWORDS,
+    # UNRESERVED_KEYWORDS,
+)
+
+from sqlfluff.core.dialects import load_raw_dialect
+
+
+ansi_dialect = load_raw_dialect("ansi")
+tsql_dialect = ansi_dialect.copy_as("tsql")
+
+
+# Bracket pairs (a set of tuples).
+# (name, startref, endref, persists)
+# NOTE: The `persists` value controls whether this type
+# of bracket is persisted during matching to speed up other
+# parts of the matching process. Round brackets are the most
+# common and match the largest areas and so are sufficient.
+ansi_dialect.sets("bracket_pairs").update(
+    [
+        ("round", "StartBracketSegment", "EndBracketSegment", True),
+        ("square", "StartSquareBracketSegment", "EndSquareBracketSegment", True),
+        ("curly", "StartCurlyBracketSegment", "EndCurlyBracketSegment", False),
+    ]
+)
+
+
+# Update only RESERVED Keywords
+# tsql_dialect.sets("reserved_keywords").clear()
+# tsql_dialect.sets("reserved_keywords").update(RESERVED_KEYWORDS)
+
+tsql_dialect.add(
+    AtSignLiteralSegment=NamedParser(
+        "atsign",
+        CodeSegment,
+        name="atsign_literal",
+        type="literal",
+        trim_chars=("@",),
+    ),
+)
+
+tsql_dialect.add(    
+    AtSignSignSegment=StringParser(
+        "@", SymbolSegment, name="atsign", type="user_designator"
+    ),   
+    SessionVariableNameSegment=RegexParser(
+        r"[@][a-zA-Z0-9_]*",
+        CodeSegment,
+        name="declared_variable",
+        type="variable",
+    ),
+)
+
+tsql_dialect.replace(
+    ParameterNameSegment=RegexParser(
+        r"[@][A-Za-z0-9_]+", CodeSegment, name="parameter", type="parameter"
+    )
+)
+
+tsql_dialect.insert_lexer_matchers(
+    [
+        RegexLexer(
+            "atsign",
+            r"[@][a-zA-Z0-9_]+",
+            CodeSegment,
+        ),
+    ],
+    before="code",
+)
+
+
+@tsql_dialect.segment(replace=True)
+class CreateFunctionStatementSegment(BaseSegment):
+    """A `CREATE FUNCTION` statement.
+
+    This version in the ANSI dialect should be a "common subset" of the
+    structure of the code for those dialects.
+    postgres: https://www.postgresql.org/docs/9.1/sql-createfunction.html
+    snowflake: https://docs.snowflake.com/en/sql-reference/sql/create-function.html
+    bigquery: https://cloud.google.com/bigquery/docs/reference/standard-sql/user-defined-functions
```

Comment:
Change this to T-SQL specific info

Reply:
Yes, it was just a draft to see part of the code